editor/inspector/assets/material.js

```diff
@@ -268,24 +301,25 @@ exports.update = async function(assetList, metaList) {
         this.dirtyData.uuid = this.asset.uuid;
         this.dirtyData.origin = '';
     }
-
+    // set this.material.technique
     this.material = await Editor.Message.request('scene', 'query-material', this.asset.uuid);
-
     // effect <select> tag
     this.$.effect.value = this.material.effect;
     setDisabled(this.asset.readonly, this.$.effect);
-
     // technique <select> tag
     this.$.technique.value = this.material.technique;
     setDisabled(this.asset.readonly, this.$.technique);
 
     this.updateTechniqueOptions();
     this.setDirtyData();
-
-    // optimize calculate speed when edit multiple materials in node mode
-    requestIdleCallback(() => {
-        this.updatePasses();
-    });
+    if (await this.getCustomInspector()) {
+        this.updateCustomInspector();
```

Comment:
if it is in a 'if case' , use function name ’hasCustomInspector‘  would be better